### ISSUE TITLE
Rancher single container v2.4.3 crashes, etcd high presure, abnormal resources usage

### ISSUE THREAD

**Issue by AndyL4N** [rancher/rancher#31425]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
not reproduced on others installations

**Result:**
Rancher server is deployed as single docker container on VM. And about 10 clusters are provisioned by it. This installation worked more than a year like a charm.  
Now  ETCD grows up to it's limit size (2GB) very fast. Usually in 5-10 minutes. Disabling all global rancher catalogs increases lifetime to 90 minutes.  Then typically we get next events:
` Failed to update lock: etcdserver: mvcc: database space exceeded`
...
`[FATAL] k3s exited with: exit status 255`

Then etcd goes in status 503.

> 2021/02/22 21:58:12 [INFO] Rancher version v2.4.3 (684884c00) is starting
2021/02/22 21:58:12 [INFO] Rancher arguments {ACMEDomains:[] AddLocal:auto Embedded:false HTTPListenPort:80 HTTPSListenPort:443 K8sMode:auto Debug:false Trace:false NoCACerts:false AuditLogPath:/var/log/auditlog/rancher-api-audit.log AuditLogMaxage:10 AuditLogMaxsize:100 AuditLogMaxbackup:10 AuditLevel:0 Features:}
2021/02/22 21:58:12 [INFO] Listening on /tmp/log.sock
2021/02/22 21:58:12 [INFO] Running etcd --data-dir=management-state/etcd
2021-02-22 21:58:12.684413 W | pkg/flags: unrecognized environment variable ETCD_URL_arm64=https://github.com/etcd-io/etcd/releases/download/v3.4.3/etcd-v3.4.3-linux-arm64.tar.gz
2021-02-22 21:58:12.684468 W | pkg/flags: unrecognized environment variable ETCD_URL_amd64=https://github.com/etcd-io/etcd/releases/download/v3.4.3/etcd-v3.4.3-linux-amd64.tar.gz
2021-02-22 21:58:12.684475 W | pkg/flags: unrecognized environment variable ETCD_UNSUPPORTED_ARCH=amd64
2021-02-22 21:58:12.684481 W | pkg/flags: unrecognized environment variable ETCD_URL=ETCD_URL_amd64
[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead
2021-02-22 21:58:12.684522 I | etcdmain: etcd Version: 3.4.3
2021-02-22 21:58:12.684528 I | etcdmain: Git SHA: 3cf2f69b5
2021-02-22 21:58:12.684533 I | etcdmain: Go Version: go1.12.12
2021-02-22 21:58:12.684537 I | etcdmain: Go OS/Arch: linux/amd64
2021-02-22 21:58:12.684543 I | etcdmain: setting maximum number of CPUs to 4, total number of available CPUs is 4
2021-02-22 21:58:12.685435 N | etcdmain: the server is already initialized as member before, starting as etcd member...
[WARNING] Deprecated '--logger=capnslog' flag is set; use '--logger=zap' flag instead
2021-02-22 21:58:12.686546 I | embed: name = default
2021-02-22 21:58:12.686557 I | embed: data dir = management-state/etcd
2021-02-22 21:58:12.686563 I | embed: member dir = management-state/etcd/member
2021-02-22 21:58:12.686568 I | embed: heartbeat = 100ms
2021-02-22 21:58:12.686573 I | embed: election = 1000ms
2021-02-22 21:58:12.686577 I | embed: snapshot count = 100000
2021-02-22 21:58:12.686607 I | embed: advertise client URLs = http://localhost:2379
2021-02-22 21:58:12.686615 I | embed: initial advertise peer URLs = http://localhost:2380
2021-02-22 21:58:12.686623 I | embed: initial cluster = 
2021-02-22 21:58:17.591266 I | etcdserver: recovered store from snapshot at index 151601521
2021-02-22 21:58:17.591889 I | mvcc: restore compact to 141084727
2021-02-22 21:58:30.426365 I | etcdserver: restarting member 8e9e05c52164694d in cluster cdf818194e3a8c32 at commit index 151624191
raft2021/02/22 21:58:30 INFO: 8e9e05c52164694d switched to configuration voters=(10276657743932975437)
raft2021/02/22 21:58:30 INFO: 8e9e05c52164694d became follower at term 105
raft2021/02/22 21:58:30 INFO: newRaft 8e9e05c52164694d [peers: [8e9e05c52164694d], term: 105, commit: 151624191, applied: 151601521, lastindex: 151624191, lastterm: 105]
2021-02-22 21:58:30.427357 I | etcdserver/api: enabled capabilities for version 3.4
2021-02-22 21:58:30.427380 I | etcdserver/membership: added member 8e9e05c52164694d [http://localhost:2380] to cluster cdf818194e3a8c32 from store
2021-02-22 21:58:30.427392 I | etcdserver/membership: set the cluster version to 3.4 from store
2021-02-22 21:58:30.429911 I | mvcc: restore compact to 141084727
2021-02-22 21:58:32.727553 W | auth: simple token is not cryptographically signed
2021-02-22 21:58:32.728682 I | etcdserver: starting server... [version: 3.4.3, cluster version: 3.4]
2021-02-22 21:58:32.732768 I | etcdserver: 8e9e05c52164694d as single-node; fast-forwarding 9 ticks (election ticks 10)
2021-02-22 21:58:32.734632 I | embed: listening for peers on 127.0.0.1:2380
raft2021/02/22 21:58:33 INFO: 8e9e05c52164694d is starting a new election at term 105
raft2021/02/22 21:58:33 INFO: 8e9e05c52164694d became candidate at term 106
raft2021/02/22 21:58:33 INFO: 8e9e05c52164694d received MsgVoteResp from 8e9e05c52164694d at term 106
raft2021/02/22 21:58:33 INFO: 8e9e05c52164694d became leader at term 106
raft2021/02/22 21:58:33 INFO: raft.node: 8e9e05c52164694d elected leader 8e9e05c52164694d at term 106
2021-02-22 21:58:35.281643 I | etcdserver: published {Name:default ClientURLs:[http://localhost:2379]} to cluster cdf818194e3a8c32
2021-02-22 21:58:35.281695 I | embed: ready to serve client requests
2021-02-22 21:58:35.284891 N | embed: serving insecure client requests on 127.0.0.1:2379, this is strongly discouraged!
2021/02/22 21:58:36 [INFO] Waiting on etcd startup: status 503
2021/02/22 21:58:37 [INFO] Waiting on etcd startup: status 503
2021/02/22 21:58:38 [INFO] Waiting on etcd startup: status 503

Manual compaction and defrag solves it but only for next hour. The investigation on our side didn't find any reasons (no infrastructure changes, no unusual deployments or configs)

Increasing CPU from 4 to 8 cores and memory from 8 to 30 GiB didn't resolve situation. 

As I see, here is very strange resource usage by k3s-server and etcd (8cores/20GiB ram config):
```
top - 21:32:18 up  4:32,  0 users,  load average: 6.79, 7.55, 7.54
Tasks:  14 total,   3 running,  11 sleeping,   0 stopped,   0 zombie
%Cpu(s): 73.5 us,  4.9 sy,  0.0 ni, 20.5 id,  0.3 wa,  0.0 hi,  0.8 si,  0.0 st
KiB Mem : 20394472 total,   192464 free, 17079932 used,  3122076 buff/cache
KiB Swap:  4194300 total,  4057876 free,   136424 used.  2999232 avail Mem 

 PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                          
   33 root      20   0 2967584 2.541g  33772 S 208.0 13.1 100:32.98 k3s-server                                                                       
    6 root      20   0 2134940 1.826g  30044 S 106.3  9.4  65:49.35 rancher                                                                          
   19 root      20   0 21.553g 0.013t 1.719g S   7.0 67.8   5:26.61 etcd         
```
![image](https://user-images.githubusercontent.com/9994053/108776775-e181c880-7573-11eb-8553-4f4e792ff823.png)
![image](https://user-images.githubusercontent.com/9994053/108777918-a2ed0d80-7575-11eb-914a-1c0771adeacd.png)

Also, the count of wal files is unusual. Normally there only 6 files, but now almost 80. And period between files is only 20s.
Contrarily, snapshots. Only five 43KB files.

I also found in rancher container logs that some errors are repeating pretty often. But not sure that it means something

> 2021/02/22 21:56:05 [ERROR] ProjectCatalogController p-7j892/rabbitmq [projectCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:46022->13.225.80.26:443: read: connection reset by peer
2021/02/22 21:56:06 [ERROR] ProjectCatalogController p-7j892/bitnami [projectCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:53384->13.225.80.4:443: read: connection reset by peer
I0222 21:56:06.319972      31 trace.go:116] Trace[721129815]: "Update" url:/apis/management.cattle.io/v3/namespaces/c-5gng7/clustercatalogs/bitnami,user-agent:Go-http-client/1.1,client:127.0.0.1 (started: 2021-02-22 21:56:05.78230486 +0000 UTC m=+278.971107550) (total time: 537.622333ms):
Trace[721129815]: [303.324208ms] [230.185928ms] Object stored in database
Trace[721129815]: [537.622333ms] [234.298125ms] END
2021/02/22 21:56:06 [ERROR] ProjectCatalogController p-h9kq9/bitnami [projectCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:49270->13.225.80.17:443: read: connection reset by peer
2021/02/22 21:56:06 [ERROR] ClusterCatalogController c-5gng7/bitnami [clusterCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:60438->13.225.80.66:443: read: connection reset by peer
2021/02/22 21:56:06 [ERROR] ProjectCatalogController p-7j892/rabbitmq [projectCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:46038->13.225.80.26:443: read: connection reset by peer
2021/02/22 21:56:07 [ERROR] ProjectCatalogController p-h9kq9/bitnami [projectCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:53400->13.225.80.4:443: read: connection reset by peer
2021/02/22 21:56:07 [ERROR] ClusterCatalogController c-5gng7/bitnami [clusterCatalog] failed with : Error in HTTP GET to [https://charts.bitnami.com/bitnami/index.yaml], error: Get https://charts.bitnami.com/bitnami/index.yaml: read tcp 172.17.0.2:60454->13.225.80.66:443: read: connection reset by peer

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
  2.4.3 (but also tested with 2.4.13 and 2.4.2)
- Installation option (single install/HA):
  Rancher single docker container

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
  N/A   
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
  On-premise VMware ESXi cluster, Virtual machine, 2.6 GHz, 4 cores, 8 GiB RAM, NVMe array storage

- Docker version (use `docker version`):

```
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        2d0083d
 Built:             Thu Jun 27 17:56:06 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       2d0083d
  Built:            Thu Jun 27 17:26:28 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by AndyL4N** [comment]
We found the cause of multiple errors of CatalogController. Some edge firewall rules were updated by Cisco subscription. It blocked all responses from bitnami.com. By the way every restart of Rancher conatainer  enables  helm3-library and git.rancher.io/charts. This really boosts cpu and memory usage (8 cores /20GiB). Disabling built-in catalogs solves problem.
![image](https://user-images.githubusercontent.com/9994053/109011550-113ee680-76c2-11eb-90fe-f34639664298.png)
![image](https://user-images.githubusercontent.com/9994053/109017543-549c5380-76c8-11eb-97ab-d8c89bc34abf.png)
